File: C++/problem0293.cpp
#include <iostream>
#include <iomanip>
#include <vector>
#include <map>
#include <set>
#include <algorithm>
#include <cmath>
#include <ctime>
#include <cstdlib>
#include <windows.h>

using namespace std;

// class Solution {
// public:
//     vector<string> generatePossibleNextMoves(string s) {
//         vector<string> ans;
//         if (s.empty())
//         {
//             return ans;
//         }
//         string tmp = s;
//         for (int k = 0; k < tmp.size()-1; ++k)
//         {
//             if ( tmp[k]=='+' && tmp[k+1]=='+' )
//             {
//                 tmp[k] = '-';
//                 tmp[k+1] = '-';
//                 ans.push_back(tmp);
//                 tmp = s;
//             }
//         }
//         return ans;
//     }
// };

class Solution {
public:
    vector<string> generatePossibleNextMoves(string s) {
        vector<string> ans;
        if (s.empty())
        {
            return ans;
        }
        string tmp = s;
        string::iterator it;
        for (it = tmp.begin(); it < tmp.end()-1; ++it)
        {
            if ( *it=='+' && *(it+1)=='+' )
            {
                *it = '-';
                *(it+1) = '-';
                ans.push_back(tmp);
                tmp = s;
            }
        }
        return ans;
    }
};

int main()
{
    string s = "++++";
    // string s = "";
    Solution solu;
    vector<string> ans = solu.generatePossibleNextMoves(s);
    vector<string>::iterator it;
    for (it = ans.begin(); it < ans.end(); ++it)
    {
        cout << *it << '\t';
    }
    cout << endl;
    return 0;
}

File: Python/problem0293.py
# -*- coding: utf-8 -*-
"""
Created on Sat Jun 29 19:34:04 2019

@author: Administrator
"""

class Solution:
    def generatePossibleNextMoves(self, s: str) -> list:
        ans = []
        tmp = list(s)
        for k in range(len(tmp)-1):
            if tmp[k] == '+' and tmp[k+1] == '+':
                tmp[k] = '-'
                tmp[k+1] = '-'
                ans.append(''.join(tmp))
                tmp = list(s)
        return ans

    
#class Solution(object):
#    def generatePossibleNextMoves(self, s):
#        """
#        :type s: str
#        :rtype: List[str]
#        """
#        res = []
#        for i in range(len(s) - 1):
#            if s[i:i+2] == "++":
#                res.append(s[:i] + "--" + s[i+2:])
#        return res
        
        
        
solu = Solution()
s = "++++"
print(solu.generatePossibleNextMoves(s))